src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -291,7 +295,8 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
             const groupResult = new AggregatedExerciseGroupResult(exerciseGroup.id, exerciseGroup.title, exerciseGroup.maxPoints, exerciseGroup.numberOfParticipants);
             // We initialize the data structure for exercises here as it can happen that no student was assigned to an exercise
             exerciseGroup.containedExercises.forEach((exerciseInfo) => {
-                const exResult = new AggregatedExerciseResult(exerciseInfo.exerciseId, exerciseInfo.title, exerciseInfo.maxPoints, exerciseInfo.numberOfParticipants);
+                const type = ExamScoresComponent.declareExerciseType(exerciseInfo);
+                const exResult = new AggregatedExerciseResult(exerciseInfo.exerciseId, exerciseInfo.title, exerciseInfo.maxPoints, exerciseInfo.numberOfParticipants, type!);
```

Comment:
I would not call it exResult, it could be exam or exercise

Reply:
It was already called exResult before, I only added a parameter to the constructor. But let's call it exerciseResult then.